internal/controller/numaflowcontrollerrollout/numaflowcontrollerrollout_controller.go

```diff
@@ -269,539 +226,312 @@ func (r *NumaflowControllerRolloutReconciler) reconcile(
 		ppnd.GetPauseModule().NewPauseRequest(controllerKey)
 	}
 
-	deployment, deploymentExists, err := r.getNumaflowControllerDeployment(ctx, controllerRollout)
+	newNumaflowControllerDef, err := generateNewNumaflowControllerDef(nfcRollout)
 	if err != nil {
-		return ctrl.Result{}, err
+		return ctrl.Result{}, fmt.Errorf("error generating NumaflowController: %v", err)
 	}
 
-	// determine the Upgrade Strategy user prefers
-	upgradeStrategy, err := usde.GetUserStrategy(ctx, controllerRollout.Namespace)
+	existingNumaflowControllerDef, err := kubernetes.GetResource(ctx, r.client, newNumaflowControllerDef.GroupVersionKind(),
```

Comment:
technically we don't need to get this as an `Unstructured` type. We could use the actual type. Did you use `Unstructured` because you were thinking you could take advantage of shared code somewhere? 

You can alternatively use `r.client.Get()` with an actual `NumaflowController`

Reply:
I see that you are calling the `usde` which takes an Unstructured type - perhaps that's the rationale for using `Unstructured`? Anything else?

Reply:
I think I'm good with it

Reply:
That is one of the reasons, I also tried to keep the code consistent between ISBSvcRollout and NumaflowControllerRollout for easier maintainability and to be able to possibly reduce code duplication at some point.

Reply:
Added comment.